python/lsst/ts/eas/m1m3_thermal.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# This file is part of ts_eas.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+
+from dds import DDSException
+from lsst.ts import salobj
+
+__all__ = [
+    "SAL_TIMEOUT",
+    "REMOTE_STARTUP_TIME",
+    "SUMMARY_STATE_TIME",
+    "FAN_SLEEP_TIME",
+    "VALVE_SLEEP_TIME",
+    "DDS_RESTART_TIME",
+    "run_control",
+]
+
+SAL_TIMEOUT = 5.0  # SAL telemetry/command timeout
+REMOTE_STARTUP_TIME = 5.0  # Time for remotes to get set up
+SUMMARY_STATE_TIME = 5.0  # Wait time for a summary state change
+FAN_SLEEP_TIME = 30.0  # Time to wait after changing the fans
+VALVE_SLEEP_TIME = 60.0  # Time to wait after changing the valve
+DDS_RESTART_TIME = 60.0  # Time to wait after a DDS exception
+
+
+async def run_loop(
```

Comment:
Is there a particular reason this code is not all part of the CSC? The fact that you have to pass around remotes is kind of a red herring. If this was all moved to the CSC you could create the remotes as CSC attributes and use it there and not have to fiddle around passing these things to methods.